nexus/tests/integration_tests/metrics.rs

```diff
@@ -690,60 +690,101 @@ async fn test_mgs_metrics(
             return;
         }
 
-        let table = timeseries_query(&cptestctx, &format!("get {metric_name}"))
-            .await
-            .into_iter()
-            .find(|t| t.name() == metric_name);
-        let table = match table {
-            Some(table) => table,
-            None => panic!("missing table for {metric_name}"),
+        let query = format!("get {metric_name}");
+
+        // MGS polls SP sensor data once every second. It's possible that, when
+        // we triggered Oximeter to collect samples from MGS, it may not have
+        // run a poll yet, so retry this a few times to avoid a flaky failure if
+        // no simulated SPs have been polled yet.
+        let result = wait_for_condition(
+            || async {
+                match check_inner(cptestctx, &metric_name, &query, &expected).await {
+                    Ok(_) => Ok(()),
+                    Err(e) => {
+                        eprintln!("{e}; will try again to ensure all samples are collected");
+                        Err(CondCheckError::<()>::NotYet)
+                    }
+                }
+            },
+            &Duration::from_secs(1),
+            // We really don't need to wait that long to know that the sensor
+            // metrics will never be present. This could probably be shorter
+            // than 30 seconds, but I want to be fairly generous to make sure
+            // there are no flaky failures even when things take way longer than
+            // expected...
+            &Duration::from_secs(30),
+        )
+        .await;
+        if result.is_err() {
+            panic!("failed to find timeseries for {query} within 60s")
```

Comment:
Might be more helpful if the message said something like, "when running query '{query}' within 30s"? It _is_ 30s, right?

Reply:
whoops, good catch! thank you!

Reply:
fixed in d2df64160c9559fa99336ad40d793c38db4ecc4a :)